src/mhd_system.hpp

```diff
@@ -26,6 +30,294 @@ template <typename problem_t> class MHDSystem
+				// extract both components of magnetic field either side of the cell-edge
+				const auto &B0_m = ec_fabs_Bi_ieside[0][0].const_array();
+				const auto &B0_p = ec_fabs_Bi_ieside[0][1].const_array();
+				const auto &B1_m = ec_fabs_Bi_ieside[1][0].const_array();
+				const auto &B1_p = ec_fabs_Bi_ieside[1][1].const_array();
+				// extract all four quadrants of the electric field about the cell-edge
+				const auto &E2_q0 = ec_fabs_E_q[0].const_array();
+				const auto &E2_q1 = ec_fabs_E_q[1].const_array();
+				const auto &E2_q2 = ec_fabs_E_q[2].const_array();
+				const auto &E2_q3 = ec_fabs_E_q[3].const_array();
+				// compute electric field on the cell-edge
+				const auto &E2_ave = ec_mf_emf_comps[wsolve][iedge_rel2face][mfi].array();
+				amrex::ParallelFor(box_ec, [=] AMREX_GPU_DEVICE(int i, int j, int k) {
+					E2_ave(i, j, k) = fspd_x0(i, j, k, 1) * fspd_x1(i, j, k, 1) * E2_q0(i, j, k) +
+							  fspd_x0(i, j, k, 0) * fspd_x1(i, j, k, 1) * E2_q3(i, j, k) +
+							  fspd_x0(i, j, k, 0) * fspd_x1(i, j, k, 0) * E2_q1(i, j, k) +
+							  fspd_x0(i, j, k, 1) * fspd_x1(i, j, k, 0) * E2_q2(i, j, k) -
+							  fspd_x1(i, j, k, 1) * fspd_x1(i, j, k, 0) / (fspd_x1(i, j, k, 1) + fspd_x1(i, j, k, 0)) *
+							      (B0_p(i, j, k) - B0_m(i, j, k)) +
+							  fspd_x0(i, j, k, 1) * fspd_x0(i, j, k, 0) / (fspd_x0(i, j, k, 1) + fspd_x0(i, j, k, 0)) *
+							      (B1_p(i, j, k) - B1_m(i, j, k));
+				});
+			}
+		}
+	}
+}
+
+template <typename problem_t>
+void MHDSystem<problem_t>::ReconstructTo(FluxDir dir, arrayconst_t &cState, array_t &lState, array_t &rState, const amrex::Box &reconstructRange,
+					 const int reconstructionOrder)
+{
+	if (reconstructionOrder == 3) {
+		switch (dir) {
+			case FluxDir::X1:
+				MHDSystem<problem_t>::template ReconstructStatesPPM<FluxDir::X1>(cState, lState, rState, reconstructRange, reconstructRange, 1);
+				break;
+			case FluxDir::X2:
+				MHDSystem<problem_t>::template ReconstructStatesPPM<FluxDir::X2>(cState, lState, rState, reconstructRange, reconstructRange, 1);
+				break;
+			case FluxDir::X3:
+				MHDSystem<problem_t>::template ReconstructStatesPPM<FluxDir::X3>(cState, lState, rState, reconstructRange, reconstructRange, 1);
+				break;
+		}
+	} else if (reconstructionOrder == 2) {
+		switch (dir) {
+			case FluxDir::X1:
+				MHDSystem<problem_t>::template ReconstructStatesPLM<FluxDir::X1, SlopeLimiter::minmod>(cState, lState, rState, reconstructRange,
+														       1);
+				break;
+			case FluxDir::X2:
+				MHDSystem<problem_t>::template ReconstructStatesPLM<FluxDir::X2, SlopeLimiter::minmod>(cState, lState, rState, reconstructRange,
+														       1);
+				break;
+			case FluxDir::X3:
+				MHDSystem<problem_t>::template ReconstructStatesPLM<FluxDir::X3, SlopeLimiter::minmod>(cState, lState, rState, reconstructRange,
+														       1);
+				break;
+		}
+	} else if (reconstructionOrder == 1) {
+		switch (dir) {
```

Comment:
This switch statement is not necessary. Make `dir` a template parameter instead.